js/tabs.js

```diff
@@ -330,11 +326,11 @@ Tabs.prototype.promptAllUnsavedFromIndex_ = function(i, callback) {
     this.showTab(this.tabs_[i].getId());
     this.promptSave_(tab, function(answer) {
       if (answer === 'yes') {
-        this.save(tab, false /* close */);
-      } else if (answer === 'cancel') {
-        return;
+        this.save(
+          tab, this.promptAllUnsavedFromIndex_.bind(this, i + 1, callback));
+      } else if (answer === 'no') {
```

Comment:
Not necessary for this patch, but I think it'd be cool to change this to an enum at some point :)

Reply:
Ack.